web/frontend/src/lib/components/robot-client.svelte

```diff
@@ -439,22 +450,20 @@ const start = () => {
   cancelTick = setAsyncInterval(tick, 500);
 };
 
-const connect = async (creds?: Credentials, authEntity?: string) => {
+const connect = async (creds?: Credential) => {
   $connectionStatus = 'connecting';
 
   let sdkCreds = undefined;
   const c = creds ?? bakedAuth.creds;
-  const ae = authEntity ?? bakedAuth.authEntity;
-  if (c && ae) {
+  if (c) {
     sdkCreds = {
       type: c.type as CredentialType,
       payload: c.payload,
-      authEntity: ae,
-    };
+      authEntity: c.authEntity ?? bakedAuth.authEntity,
+    } as Credential;
```

Comment:
[nit] use `satisfies Credential` here or add more type info to `sdkCreds`

Reply:
tyty